python/paddle/fluid/params.py

```diff
@@ -0,0 +1,39 @@
+#   Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from . import core
+
+__all__ = ['Params', ]
+
+
+class Params(object):
+    def __init__(self, path=None):
+        self.scope = core.Scope()
+
+        if path:
+            self._load(path)
+
+    def _load(self, path):
+        # reference: load_persistables in io.py
+        pass
+
+    def save(self, path):
+        # reference: save_persistables in io.py
+        pass
+
+    def add_params(self, scope):
```

Comment:
I don't quite understand here -- the method name is *add params*, but the method comment says that it *takes keys from the scope*. Does it really add something?

Reply:
@jetfuel maybe we can implement merge in the C++? how do you think?

Reply:
Fixed in https://github.com/PaddlePaddle/Paddle/pull/10354

Reply:
Every var in the `scope` argument will be put into the receiving scope, if the receiving scope already has the same name, it will be replaced. So I guess we can just do:
```C++
void Scope::merge(Scope* scope) const {
  std::vector<std::string> names = scope.LocalVarNames;
  for (auto name_it = names.begin(); name_it != names.end(); name_it ++) {
    vars_[*name_it] = scope.Var(*name_it)
  }
}
```

Sorry the spec was different from what we discussed yesterday (yesterday's was to not replace if already exists).